netty/src/test/java/io/grpc/netty/NettyHandlerTestBase.java

```diff
@@ -94,12 +113,50 @@ protected final void initChannel(Http2HeadersDecoder headersDecoder) throws Exce
 
     handler = newHandler();
 
-    channel = new EmbeddedChannel(handler);
+    channel = new FakeClockSupportedChanel(handler);
     ctx = channel.pipeline().context(handler);
 
     writeQueue = initWriteQueue();
   }
 
+  private final class FakeClockSupportedChanel extends EmbeddedChannel {
+    EventLoop eventLoop;
+
+    FakeClockSupportedChanel(ChannelHandler... handlers) {
+      super(handlers);
+    }
+
+    @Override
+    public EventLoop eventLoop() {
+      if (eventLoop == null) {
+        createEventLoop();
+      }
+      return eventLoop;
+    }
+
+    void createEventLoop() {
+      EventLoop realEventLoop = super.eventLoop();
+      if (realEventLoop == null) {
+        return;
+      }
+      eventLoop = mock(EventLoop.class, delegatesTo(realEventLoop));
+      ArgumentMatcher<Runnable> isFakeClockRunnable = new ArgumentMatcher<Runnable>() {
+        @Override
+        public boolean matches(Object argument) {
+          String name = argument.toString();
+          for (String taskName : scheduleTheseWithFakeClock) {
+            if (name.contains(taskName)) {
+              return true;
+            }
+          }
+          return false;
+        }
+      };
+      doAnswer(delegatesTo(fakeClock.getScheduledExecutorService())).when(eventLoop)
+          .schedule(argThat(isFakeClockRunnable), anyLong(), any(TimeUnit.class));
```

Comment:
I'm not sure if it's safe to do so, because I don't how much Netty code calls on the real EmbeddedEventLoop directly. The stubbed eventLoop will only work if Netty code is calling on the value of `channel.eventLoop()`. For example, if Netty schedules a task with the value of `channel.eventLoop()` and calls `runPendingTasks()` on the real EmbeddedEventLoop, the task will not run.